fastapi/_compat.py

```diff
@@ -494,12 +499,21 @@ def get_definitions(
         )
 
     def is_scalar_field(field: ModelField) -> bool:
+        if (inner := root_model_inner_type(field.type_)) is not None:
```

Comment:
_compat checks in V1/V2 work quite differently and I'm not sure is it worth trying to support legacy Pydantic. Could just remove these